brightray/common/application_info_win.cc

```diff
@@ -62,4 +65,20 @@ bool GetAppUserModelID(ScopedHString* app_id) {
   return app_id->success();
 }
 
+bool IsRunningInDesktopBridge() {
+  if (hasCheckedIsRunningInDesktopBridge) {
```

Comment:
You could do this instead
```
static bool result = IsRunningInDesktopBridgeImpl();
return result;
```

And you don’t need those 2 global variables

Reply:
[`std::call_once`](http://en.cppreference.com/w/cpp/thread/call_once) would be more expressive, though at the cost of keeping the two global variables.

```cpp
#include <mutex>

bool IsRunningInDesktopBridge() {
  static std::once_flag once;
  static bool bridge = false;

  std::call_once(&once, [&bridge]{
    UINT32 length;
    wchar_t packageFamilyName[PACKAGE_FAMILY_NAME_MAX_LENGTH + 1];
    HANDLE proc = GetCurrentProcess();
    LONG result = GetPackageFamilyName(proc, &length, packageFamilyName);
    bridge = result == ERROR_SUCCESS;
  });

  return bridge;
}￼
```

It doesn't look to me like IsRunningInDesktopBridge() would be harmed by a race condition, so the main advantage of this is the builtin expressiveness of using std:: types that indicate one flag is used to bootstrap the other variable.